pkg/hooks/audit_logger_sqlite.go

```diff
@@ -106,6 +112,27 @@ func (al *SQLiteAuditLogger) Close() error {
 	return al.sqlite.Close()
 }
 
+func (al *SQLiteAuditLogger) withRetries(f func() error, n int) error {
+	var err error
+	err = f()
```

Comment:
is there something that makes it so that this duplicate of 124-128 can't be removed? Code is the same in the `for` loop! Would just need to start attempts at 0.